src/compiler/types.ts

```diff
@@ -5298,7 +5298,7 @@ export const enum TypeFormatFlags {
     None                                    = 0,
     NoTruncation                            = 1 << 0,  // Don't truncate typeToString result
     WriteArrayAsGenericType                 = 1 << 1,  // Write Array<T> instead T[]
-    // hole because there's a hole in node builder flags
+    GenerateNamesForShadowedTypeParams      = 1 << 2,   // When a type parameter T is shadowing another T, generate a name for it so it can still be referenced

```

Comment:
This API change is to be able to use this flag via the type printing system, since TypeFormatFlags and NodeBuilderFlags overlap intentionally.

Reply:
I am not sure if changing this option in baselines is good or not since Quickinfo doesnt use this flag and thats what our type baselines go for?

Reply:
Honestly I was thinking about whether or not this should be the default for quick info; it too can cause confusion as in: https://github.com/microsoft/TypeScript/pull/55820#issuecomment-1731378414